Sources/WalletConnectPush/Client/Wallet/PushSyncService.swift

```diff
@@ -23,4 +41,72 @@ final class PushSyncService {
             throw PushError.registerSignatureRejected
         }
     }
+
+    func fetchHistoryIfNeeded(account: Account) async throws {
+        guard isColdStart(account: account) else { return }
+
+        try await historyClient.register(tags: [
+            "5000", // sync_set
+            "5002", // sync_delete
+            "4002"  // push_message
+        ])
+
+        let syncTopic = try subscriptionsStore.getStoreTopic(account: account)
+
+        let updates: [StoreSetDelete] = try await historyClient.getMessages(
+            topic: syncTopic,
+            count: 200,
+            direction: .backward
+        )
+
+        let inserts: [PushSubscription] = updates.compactMap { update in
+            guard let value = update.value else { return nil }
+            return try? JSONDecoder().decode(PushSubscription.self, from: Data(value.utf8))
+        }
+
+        let deletions: [String] = updates.compactMap { update in
+            guard update.value == nil else { return nil }
+            return update.key
+        }
+
+        let subscriptions = inserts.filter { !deletions.contains( $0.databaseId ) }
+
+        try subscriptionsStore.setInStore(objects: subscriptions, for: account)
+
+        for subscription in subscriptions {
+            let symmetricKey = try SymmetricKey(hex: subscription.symKey)
+            try kms.setSymmetricKey(symmetricKey, for: subscription.topic)
+            try await networkingInteractor.subscribe(topic: subscription.topic)
+
+            let historyRecords: [HistoryRecord<PushMessage>] = try await historyClient.getRecords(
+                topic: subscription.topic,
+                count: 200,
+                direction: .backward
+            )
+
+            let messageRecords = historyRecords.map { record in
+                return PushMessageRecord(
+                    id: record.id.string,
+                    topic: subscription.topic,
+                    message: record.object,
+                    publishedAt: Date()
```

Comment:
can we get a date from a history somehow? in w3i there is a label that displays the date. With this implementation the date will be incorrect.